dc/s2n-quic-dc/src/path/secret/map/rehandshake.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use rand::seq::SliceRandom;
+use std::{
+    net::SocketAddr,
+    time::{Duration, Instant},
+};
+
+pub(super) struct RehandshakeState {
+    // FIXME: This is larger than it needs to be because SocketAddr is 32 bytes. We should consider
+    // some other storage form, since IPv4 is 6 bytes and IPv6 is 28 bytes (18 bytes if we ignore
+    // scope ID), almost anything would be smaller than this. But this is cheaper to implement and
+    // we can revisit memory impact separately.
+    //
+    // Splitting IPv4 and IPv6 would help, but it would also mean that we scan one and then the
+    // other, which is probably bad idea long-term -- we want to visit peers randomly.
+    queue: Vec<SocketAddr>,
```

Comment:
We could use [SocketAddressV6](https://docs.rs/s2n-quic-core/latest/s2n_quic_core/inet/ipv6/struct.SocketAddressV6.html) and do a [V4 mapped addresses](https://docs.rs/s2n-quic-core/latest/s2n_quic_core/inet/ipv6/struct.SocketAddressV6.html#method.unmap). The nice thing is that struct is unaligned because it supports zero copy so it should pack pretty well. Not blocking though.